File: DiscordStubs/src/main/java/com/discord/models/commands/ApplicationCommandOption.java
package com.discord.models.commands;

import com.discord.api.commands.ApplicationCommandType;
import com.discord.api.commands.CommandChoice;

import java.util.List;

@SuppressWarnings("unused")
public final class ApplicationCommandOption {
    public ApplicationCommandOption(
            ApplicationCommandType type,
            String name,
            String description,
            Integer descriptionRes,
            boolean required,
            boolean def,
            List<CommandChoice> choices,
            List<ApplicationCommandOption> options
    ) {}

    public final String getName() { return ""; }
}

File: Aliucord/src/main/java/com/aliucord/api/kotlin/data/ApplicationCommandOption.kt
package com.aliucord.api.kotlin.data

import com.discord.api.commands.ApplicationCommandType
import com.discord.api.commands.CommandChoice
import com.discord.models.commands.ApplicationCommandOption

fun applicationCommandOption(
    type: ApplicationCommandType,
    name: String,
    description: String,
    descriptionRes: Int? = null,
    required: Boolean = false,
    def: Boolean = false,
    choices: List<CommandChoice>? = null,
    options: List<ApplicationCommandOption>?  = null
) = ApplicationCommandOption(
    type,
    name,
    description,
    descriptionRes,
    required,
    def,
    choices,
    options
)